### PR TITLE
UGENE-7003. Invalid Python module validation in Windows 10

### PR DESCRIPTION
**Pull request by evelinabis**
Changed Python module validation message

### CODE REVIEW

File: src/plugins/external_tool_support/src/python/PythonSupport.cpp [hunk 1]
```diff
@@ -53,11 +53,13 @@ PythonSupport::PythonSupport()
     executableFileName = "python2.7";
 #    endif
 #endif
-    validMessage = "Python ";
+    const QString pythonVersionRegexpStr = "(\\d+.\\d+.\\d+)";
```
Comment (DmitriiSukhomlinov) [inline src/plugins/external_tool_support/src/python/PythonSupport.cpp]: `static const QString PYTHON_VERSION_REGEXP = "(\\d+.\\d+.\\d+)";`

**Review by DmitriiSukhomlinov — DISMISSED** [review]

**Review by yalgaer — CHANGES_REQUESTED** [review]
This change can be tested with GUI tests
1) Add an executable file to the SVN or create inside of the test. (I would prefer adding to the SVN a file like 'dumb-script.sh/cmd' that can be used to test different external tools
2) In the test ensure that 'UGENE_EXTERNAL_TOOLS_VALIDATION_BY_PATH_ONLY' is not set to "1"
3) Add the 'dumb.sh' as a Python executable.
4) Check that validation fails.

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    CHECK_SET_ERR(qgetenv("UGENE_EXTERNAL_TOOLS_VALIDATION_BY_PATH_ONLY") != "1",
+                  "UGENE_EXTERNAL_TOOLS_VALIDATION_BY_PATH_ONLY is set to 1, but expected 0")
+
+    class CheckPythonInvalidation : public CustomScenario {
+        void run(GUITestOpStatus &os) override {
+            AppSettingsDialogFiller::openTab(os, AppSettingsDialogFiller::ExternalTools);
+
+            QString toolPath = testDir + "_common_data/regression/7003/dumb.";
+            #ifdef Q_OS_WIN
```
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: nit: there is a more readatable API:  toolPath+=isOsWindows()?"a":"b"

**Review by yalgaer — COMMENTED** [review]
Everything is OK, please show that the test runs correctly in the Teamcity and I will approve.

+ Looks like you need to merge with master. The h/cpp file with regression test was added in a parallel PR.

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp [hunk 3]
```diff
@@ -19,21 +19,59 @@
...
 namespace GUITest_regression_scenarios {
 using namespace HI;
 
+GUI_TEST_CLASS_DEFINITION(test_7003) {
+    // 1. Ensure that 'UGENE_EXTERNAL_TOOLS_VALIDATION_BY_PATH_ONLY' is not set to "1"
+    // 2. Open "UGENE Application Settings", select "External Tools" tab
+    // 3. Add the 'dumb.sh' or 'dumb.cmd' as a Python executable
+    // 4. Check that validation fails
+
+    CHECK_SET_ERR(qgetenv("UGENE_EXTERNAL_TOOLS_VALIDATION_BY_PATH_ONLY") != "1",
```
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: You need to reset the variable in the test if it was set externally:
qputenv("UGENE_EXTERNAL_TOOLS_VALIDATION_BY_PATH_ONLY", "0");

**Review by yalgaer — CHANGES_REQUESTED** [review]

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp [hunk 4]
```diff
@@ -19,25 +19,66 @@
...
+                          "Python module is expected to be invalid, but in fact it is valid")
+
+            GTUtilsDialog::clickButtonBox(os, GTWidget::getActiveModalWidget(os), QDialogButtonBox::Cancel);
+        }
+    };
+
+    GTUtilsDialog::waitForDialog(os, new AppSettingsDialogFiller(os, new CheckPythonInvalidation()));
+    GTMenu::clickMainMenuItem(os, QStringList() << "Settings"
+                                                << "Preferences...",
+                              GTGlobals::UseMouse);
```
Comment (DmitriiSukhomlinov) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: Offset

**Comment by evelinabis** [comment]
@yalgaer, @atiunov anyone want to approve me?

**Review by yalgaer — DISMISSED** [review]

**Review by yalgaer — APPROVED** [review]

**Pull request merged**